runtime/common/NoiseModel.h

```diff
@@ -297,4 +298,38 @@ class phase_flip_channel : public kraus_channel {
     validateCompleteness();
   }
 };
+
+/// @brief readout_error is a type of error that Models
+/// the probabilistic errors that can happen when measuring qubits.
+/// It allows to define a custom probability distribution that describes how
+/// often a qubit's measurement result differs from its actual state.
+/// Its constructor expects a 2x2 matrix of float values that represents the
+/// probability matrix applied to the probabilistic state of the qubit measurement.
+/// It takes the form of
+/// P = [[p(0|0), p(0|1)], [p(1|0), p(1,1)]]
+/// where p(i|j) is the probability of measuring outcome i when the actual state is j
+class readout_error : public kraus_channel {
+public:
+  readout_error(std::vector<cudaq::real> data) : kraus_channel() {
+    ops = probMatrixToKraus(data);
+    validateCompleteness();
+  }
+  template <typename T>
+  readout_error(std::initializer_list<T> &&initList) {
+    ops = probMatrixToKraus(initList);
+    validateCompleteness();
+  }
+private:
+  std::vector<kraus_op>
+  probMatrixToKraus(const std::vector<cudaq::real>& probMatrix) {
+    std::vector<cudaq::complex> k0 = {std::sqrt(probMatrix[0]), 0,
```

Comment:
I saw that the real numbers are on 128 bits (long double). std::sqrt is only for 64 bits (double). Maybe we should use std::sqrtl.

Reply:
we are not using long double, std::real is using simple double